### PR TITLE
add signed commits RFC to docs

### PR DESCRIPTION
**Pull request by JanMa**
Signed-off-by: Jan Martens <jan@martens.eu.org>

Closes: #399

### CODE REVIEW

File: website/content/docs/rfcs/signed-commits.mdx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  access to necessary tools are discouraged from contributing, which reduces
+  community engagement and diversity.
+* **Temporal Validity of Verification**: The validity of commit signatures is
+  temporal. If a contributor deletes their SSH key from GitHub's UI, previously
+  "verified" commits may become "unverified," compromising the long-term
+  reliability of signature verification.
+* **Rebase Merge Issues**: Ideally, we would like to rebase pull requests when
+  merging to maintain the original commit history of the PR. However, due to a
+  long-standing bug in GitHub, all signatures in individual commits become
+  invalidated during a rebase, causing them to appear as unverified.
```
Comment (cipherboy) [inline website/content/docs/rfcs/signed-commits.mdx:41]: One minor additional nit is that we use squash merges, which re-signs the squashed version of the original commits in the series with GitHub's key, effectively deleting the contributor's own attestation.

File: website/content/docs/rfcs/signed-commits.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+## Technical Description
+To implement this change, we will:
+
+1. Update the branch protection rules in GitHub to remove the requirement for
+   signed commits.
+2. Add a GitHub Actions Workflow to ensure Maintainer commits are always
+   verified.
+2. Communicate the change to all contributors through appropriate channels
+   (e.g., mailing lists, GitHub discussions).
+3. Update our contribution guidelines to reflect the new policy.
```
Comment (cipherboy) [inline website/content/docs/rfcs/signed-commits.mdx:61]: FWIW, our existing [CONTRIBUTING.md](https://github.com/openbao/openbao/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-dco-sign-off-policy) only mentions DCO and not commit GPG signing, so formalizing this policy actually brings us in line with the existing guidelines. :-)

**Review by cipherboy — APPROVED** [review]
Much appreciated, thanks @JanMa! I don't think the discussion on squash merges is worth holding up the merge over, but is otherwise useful to point out.

File: website/content/docs/rfcs/signed-commits.mdx (hunk 1 above)
Comment (JanMa) [reply website/content/docs/rfcs/signed-commits.mdx:41]: It is mentioned a bit below in the `Rationale` section that we use Squash Merge

Comment (cipherboy) [reply website/content/docs/rfcs/signed-commits.mdx:41]: Oh, d'oh. ;-) Thanks! :D

**Pull request merged**